so it seems to me like the failures are due to `lirc` which for some reason builds on hydra correctly but when trying `nix-build -A lirc --check` on master fails.

```
make[3]: Nothing to be done for 'install-exec-am'.
 /nix/store/b1wvkjx96i3s7wblz38ya0zr8i93zbc5-coreutils-9.5/bin/mkdir -p '/lib/systemd/system'
/nix/store/b1wvkjx96i3s7wblz38ya0zr8i93zbc5-coreutils-9.5/bin/mkdir: cannot create directory '/lib': Permission denied
make[3]: *** [Makefile:370: install-dist_systemdsystemunitDATA] Error 1
```

but on hydra it is creating a directory in the nix store.
https://cache.nixos.org/log/s5d104skq1w1f611rd784xnd2jnmq8mc-lirc-0.10.2.drv

anyway -- doesn't seem like it has anything to do with this change or `staging-next` unfortunately.

--- a/pkgs/development/libraries/libftdi/1.x.nix
+++ b/pkgs/development/libraries/libftdi/1.x.nix
@@ -1,6 +1,7 @@
 { lib
 , stdenv
 , fetchgit
+, fetchpatch
 , cmake
 , pkg-config
 , libusb1
@@ -28,6 +29,16 @@ stdenv.mkDerivation rec {
     rev = "de9f01ece34d2fe6e842e0250a38f4b16eda2429";
     hash = "sha256-U37M5P7itTF1262oW+txbKxcw2lhYHAwy1ML51SDVMs=";
   };
+
+  patches = [
+    (fetchpatch {
+      # http://developer.intra2net.com/mailarchive/html/libftdi/2024/msg00024.html
+      # https://bugzilla.redhat.com/show_bug.cgi?id=2319133
+      name = "swig-4.3.0-fix.patch";
+      url = "https://src.fedoraproject.org/rpms/libftdi/raw/9051ea9ea767eced58b69d855a5d700a5d4602cc/f/libftdi-1.5-swig-4.3.patch";
+      hash = "sha256-X5tqiPewnyAyvLzR6s0VbNpZKLd0idtPGU4ro36CZHI=";
+    })
+  ];
 
   strictDeps = true;
 

--- a/pkgs/development/libraries/libftdi/1.x.nix
+++ b/pkgs/development/libraries/libftdi/1.x.nix
@@ -70,14 +70,6 @@ stdenv.mkDerivation rec {
     cp -r doc/html "$out/share/doc/libftdi1/"
   '';
 
-  postFixup = optionalString cppSupport ''
-    # This gets misassigned to the C++ version's path for some reason
-    for fileToFix in $out/{bin/libftdi1-config,lib/pkgconfig/libftdi1.pc}; do
-      substituteInPlace $fileToFix \
-        --replace "$out/include/libftdipp1" "$out/include/libftdi1"
-    done
-  '';
-
   meta = with lib; {
     description = "Library to talk to FTDI chips using libusb";
     homepage = "https://www.intra2net.com/en/developer/libftdi/";

